rs/ethereum/cketh/minter/src/deposit.rs

```diff
@@ -403,3 +178,183 @@ pub async fn update_last_observed_block_number() -> Option<BlockNumber> {
         }
     }
 }
+
+async fn scrape_until_block<S: LogScraping, P: LogParser>(
```

Comment:
Good idea! Since the concrete type of `LogScraping` knows which parser should be used, I formalized it as an associated type on `LogScraping`. I was ensure whether associated types could be generic, but apparently that's the case since a while (see [generic associated types](https://rust-lang.github.io/rfcs/1598-generic_associated_types.html) if interested). PTAL

Reply:
I will close this thread to merge, but don't hesitate to let me know if you had something else in mind.